### ISSUE TITLE
Add a dedicated ShieldGemma class

### ISSUE THREAD

**Issue by RyanMullins** [keras-team/keras-hub#1973]
**Is your feature request related to a problem? Please describe.**

ShieldGemma was released in Jul 2024 with [v0.14.3](https://github.com/keras-team/keras-hub/releases/tag/v0.14.3), but since then we've had feedback that the [Example Use](https://www.kaggle.com/models/google/shieldgemma#example-use) on the model card is prohibitively complex. 

**Describe the solution you'd like**

I want to make a `keras_hub.models.ShieldGemma` class that encapsulates the functionality from the [Example Use](https://www.kaggle.com/models/google/shieldgemma#example-use) so that a user can load and use the model in two lines:

```python
shieldgemma = keras_hub.models.ShieldGemma.from_preset("shieldgemma_2b_en")
shieldgemma.predict(inputs)
```

**Describe alternatives you've considered**

The primary alternative is to use ShieldGemma via the existing method from the [Example Use](https://www.kaggle.com/models/google/shieldgemma#example-use).

**Additional context**

What remains unclear to me is precisely how to implement the `keras_hub.models.ShieldGemma` class. Should it be a subclass of [`keras.Model`](https://keras.io/api/models/model/#by-subclassing-the-model-class), [`keras_hub.models.Task`](https://keras.io/api/keras_nlp/base_classes/task/), [`keras_hub.models.TextClassifier`](https://keras.io/api/keras_nlp/base_classes/text_classifier/), or something else entirely? 

I think the uncertainty here is that ShieldGemma is acting like a `keras_hub.models.TextClassifier` by using [Gemma-as-a-Judge](https://proceedings.neurips.cc/paper_files/paper/2023/hash/91f18a1287b398d378ef22505bf41832-Abstract-Datasets_and_Benchmarks.html), but the architecture isn't quite what you could expect from a conventional classifier, like BERT. Instead of a dense layer over a limited vocabulary, the classification probability is computed from the logits for two specific tokens in the vocabulary.

**Comment by RyanMullins** [comment]
Tagging @mattdangerw who helped with the ShieldGemma launch.
